language/stdlib/modules/event.mvir

```diff
@@ -58,33 +58,33 @@ module Event {
     }
 
     // Use the sender's HandleIdGenerator to generate a unique event handle that one can emit an event to.
-    public new_event_handle(): Self.Handle acquires HandleIdGenerator {
-        return Handle {counter: 0, guid: Self.fresh_guid()};
+    public new_event_handle<T>(): Self.Handle<T> acquires HandleIdGenerator {
+        return Handle<T> {counter: 0, guid: Self.fresh_guid()};
     }
 
     // Emit an event with payload `msg` by using handle's key and counter. Will change the payload from bytearray to a
     // generic type parameter once we have generics.
-    public emit_event(handle_ref: &mut Self.Handle, msg: bytearray) {
+    public emit_event<T>(handle_ref: &mut Self.Handle<T>, msg: T) {
         let count: &mut u64;
         let guid: bytearray;
 
         guid = *&copy(handle_ref).guid;
         count = &mut move(handle_ref).counter;
 
-        Self.write_to_event_store(move(guid), *copy(count), move(msg));
+        Self.write_to_event_store<T>(move(guid), *copy(count), move(msg));
         *move(count) = *copy(count) + 1;
         return;
     }
 
     // Native procedure that writes to the actual event stream in Event store
     // This will replace the "native" portion of EmitEvent bytecode
-    native write_to_event_store(guid: bytearray, count: u64, msg: bytearray);
+    native write_to_event_store<T>(guid: bytearray, count: u64, msg: T);
```

Comment:
This needs to be `T: unrestricted`. Otherwise, any resource type `R` could be destroyed by creating an `EventHandle<R>`, then calling `Event.emit_event(handle_ref, some_r)`.

We should add a comment emphasizing the importance of not allow a resource type parameter here + add a test.